wifistats/index.js

```diff
@@ -0,0 +1,99 @@
+/* global process */
+const exec          = require('child_process').exec;
+
+// The tools
+const airport       = require('./airport');
+const procwireless  = require('./procwireless');
+const netsh         = require('./netsh');
+
+let toolInstance;
+let supportWifiStats = true;
+
+/**
+ * Uses all available tools to query for wifi stats, and the one we found to
+ * work we store in global variable {@link toolInstance}, otherwise we set
+ * supportWifiStats to false.
+ * We call all tools in parallel and when all finishes we check the results.
+ *
+ * @return {Promise}
+ */
+function initTools() {
+    const os = process.platform;
+    let tool;
+
+    // check for running OS and use appropriate tool
+    if (os === 'win32') {
+        tool = netsh;
+    } else if (os === 'darwin') {
+        tool = airport;
+    } else if (os === 'linux') {
+        tool = procwireless;
+    }
+
+    return new Promise((resolve, reject) => {
```

Comment:
If getStats is already returning a promise I don't think you have to wrap this in an additional promise. Instead you can return getStates().then(...).catch(...). Just make sure to return result from the .then and maybe return a Promise.reject(error) from the catch.